### PR TITLE
feat: added advanced_options_config

### LINKED ISSUE #25: Support for advanced_options_config
Thank you the module. Any plans on adding the json parsing block? 
Ref: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_security_policy#advanced_options_config

### PR DESCRIPTION
**Pull request by imrannayer**
fix #25

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -27,9 +27,9 @@ terraform {
     }
   }
   provider_meta "google" {
-    module_name = "blueprints/terraform/terraform-google-cloud-armor/0.1.0"
+    module_name = "blueprints/terraform/terraform-google-cloud-armor/0.3.0"
```
Comment (g-awmalik) [inline versions.tf:30]: Missing a "v" in the version no. i.e.:

`module_name = "blueprints/terraform/terraform-google-cloud-armor/v0.3.0"`

Same for the one below

File: build/int.cloudbuild.yaml [hunk 2]
```diff
@@ -21,21 +21,48 @@ steps:
...
+  waitFor:
+    - security-policy-all-verify
   name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
   args: ['/bin/bash', '-c', 'cft test run TestSecurityPolicy --stage teardown --verbose']
+
+- id: security-policy-simple-apply
+  waitFor:
+    - init-all
+  name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
+  args: ['/bin/bash', '-c', 'cft test run TestAll/examples/simple-example --stage apply --verbose']
```
Comment (g-awmalik) [inline build/int.cloudbuild.yaml:51]: Do we need a golang test for this? Do we not have any assertions as part of verify?

Comment (imrannayer) [reply build/int.cloudbuild.yaml:51]: All the tests are in examples/security-policy-all where I have all the assertions to verify everything. This is just a different variation with simple example. Just added to the test to make sure it passes the test.

Comment (imrannayer) [reply build/int.cloudbuild.yaml]: Added test and assertions for simple example

File: versions.tf [hunk 3]
```diff
@@ -27,9 +27,9 @@ terraform {
     }
   }
   provider_meta "google" {
-    module_name = "blueprints/terraform/terraform-google-cloud-armor/0.1.0"
+    module_name = "blueprints/terraform/terraform-google-cloud-armor/v0.3.0"
   }
   provider_meta "google-beta" {
-    module_name = "blueprints/terraform/terraform-google-cloud-armor/0.1.0"
+    module_name = "blueprints/terraform/terraform-google-cloud-armor/v0.3.0"
```
Comment (g-awmalik) [inline versions.tf:33]: Also, you don't need to increment this yourself. release-please will update this automatically when it kicks off a new release build.

**Review by g-awmalik — APPROVED** [review]

**Pull request merged**